Explain new token exchange strategy (#343)
SAP/cloud-sdk: docs/java/features/connectivity/destinations.mdx

Here is an example for `SUBSCRIBER_THEN_PROVIDER` option:

```java
DestinationOptions options =
    DestinationOptions
        .builder()
        .augmentBuilder(
            ScpCfDestinationOptionsAugmenter
                .augmenter()
                .retrievalStrategy(ScpCfDestinationRetrievalStrategy.SUBSCRIBER_THEN_PROVIDER))
        .build();
```

Besides the retrieval strategy, the SAP Cloud SDK also offers a _token exchange strategy_ that lets you change the behavior when authentication requires a user token exchange.
There are three different options available:

- `LOOKUP_THEN_EXCHANGE` (default)
- `LOOKUP_ONLY`
- `EXCHANGE_ONLY`

> By default, the SAP Cloud SDK first performs a "look up" request against the destination service, which yields the destination.
> This is needed to get the authentication type configured in the destination, which might then require a user token exchange.

Reviewer: ```suggestion
By default, the SAP Cloud SDK first looks up the destination on the destination service and evaluates the authentication type.
If the authentication type requires a user token exchange the SAP Cloud SDK immediately performs that exchange.
```

> In special cases, for example, if you know that the destination you are about to retrieve requires a user token exchange, performance can be improved by skipping the initial "look up" request like so:

Reviewer: ```suggestion
If you know that the destination you are about to retrieve requires a user token exchange you can improve performance by skipping the initial "look up" request like so:
```
